### PR TITLE
Ability to supply product flavor for `react-native run-android` command

### PR DESCRIPTION
**Pull request by alwx**
This small update to runAndroid.js allows to specify product flavor as optional argument, like that:
`react-native run-android --option-flavor=staging`

This option is useful when developing complex applications that require some flavor-specific functionality. More information about productFlavors can be found here: http://developer.android.com/intl/ru/tools/building/configuring-gradle.html

### CODE REVIEW

**Comment by martinbigio** [comment]
cc @foghina @mkonicek

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -64,7 +64,15 @@ function buildAndRun(args, reject) {
...
       : './gradlew';
 
-    const gradleArgs = ['installDebug'];
+    const gradleArgs = [];
+    if (args['option-flavor']) {
+        gradleArgs.push("install" +
+          args['option-flavor'][0].toUpperCase() + args['option-flavor'].slice(1)
+        );
+    } else {
+        gradleArgs.push("installDebug");
```
Comment (mkonicek) [inline local-cli/runAndroid/runAndroid.js]: nit: Use single quotes for consistency with the rest of the file.

**Comment by mkonicek** [comment]
Looks good in general, thanks!

Should the arg be added to the `parseCommandLine` call (beginning of `_runAndroid`)?

**Comment by alwx** [comment]
@mkonicek I've updated my PR. Please, check it again.
Thanks!

**Comment by alwx** [comment]
@mkonicek  I've updated my PR. Please, check it again.
Thanks!

**Comment by mkonicek** [comment]
One last small thing I realized: is there are a reason to call it `--option-flavor` rather than `--flavor`?

How about `react-native run-android --flavor=staging`? Would that make sense?

Sorry for the back and forth on this.

**Comment by alwx** [comment]
@mkonicek done.

**Comment by alwx** [comment]
@mkonicek I've updated it. Please, check it again.

**Comment by mkonicek** [comment]
Thanks for the ping @alwx!

@facebook-github-bot shipit

**Pull request closed without merging**